sdk/timeseriesinsights/Azure.Iot.TimeSeriesInsights/design/ModelSettings_API_Design.md

```diff
@@ -0,0 +1,75 @@
+# Time Series Insights
+
+## Model Settings
+
+The main motivation for Time Series Model is to simplify finding and analyzing IoT or Time Series data. It achieves this objective by enabling the curation, maintenance, and enrichment of time series data to help prepare consumer-ready datasets for analytics.
+
+Time Series Model settings can be managed through the Model Settings API which includes model display name, Time Series ID properties and default type ID. Every Gen2 environment has a model that is automatically created.
+
+### GET /timeseries/modelSettings
+
+Returns the model settings which includes model display name, Time Series ID properties and default type ID. Every Gen2 environment has a model that is automatically created.
+
+```javascript
+{
+  "parameters": {
+    "api-version": "2020-07-31",
+    "environmentFqdn": "10000000-0000-0000-0000-100000000109.env.timeseries.azure.com"
+  },
+  "responses": {
+    "200": {
+      "body": {
+        "modelSettings": {
+          "name": "DefaultModel",
+          "timeSeriesIdProperties": [
+            {
+              "name": "DeviceId",
+              "type": "String"
+            }
+          ],
+          "defaultTypeId": "5AB70D71-A8CD-410E-B70D-6F04AB9C132C"
+        }
+      }
+    }
+  }
+}
+```
+
+```csharp
+public virtual async Task<Response<ModelSettingsResponse>> GetAsync(string clientSessionId = null, CancellationToken cancellationToken = default)
```

Comment:
What is the meaning of the optional `clientSessionId` parameter to each API?

Reply:
This is what I found in TSI documentation:
A client session ID. The service records this value. Allows the service to trace a group of related operations across services.

That reminds me to add headers, I will do that now.

Reply:
That sounds like distributed tracing. IIRC there is a different way to do that. Please check with Azure SDK team on guidance.

Reply:
I asked the TSI team about clientSessionId. @yeskarthik @rasidhan can keep me honest here.

The clientSessionId is an Id used to track a single user session in the UX. So multiple operations can belong to a single client sessions Id. This is different than the client request Id where Azure core takes care of that in the framework itself. I checked with Raghav and Karthik, clientSessionId is something we'd want to expose in the client library.